framework/Optimizers/Optimizer.py

```diff
@@ -810,7 +810,7 @@ def getLossFunctionGivenId(self, evaluationID):
       @ In, evaluationID, string, the evaluation identifier (prefix)
       @ Out, functionValue, float, the loss function value
     """
-    objective  = self.mdlEvalHist.getParametersValues('outputs', nodeId = 'RecontructEnding')[self.objVar]
+    objective  = self.mdlEvalHist.getVars('output', nodeId = 'RecontructEnding')[self.objVar]
```

Comment:
we do not have "ReconstructEnding" in getVars

Reply:
Yes, the Optimizer is still in the works.  I accidentally included parts of it, but I have it in separate work.